src/test/java/com/opentok/test/OpenTokTest.java

```diff
@@ -681,29 +676,249 @@ public void testListArchives() throws OpenTokException {
+                findAll(getRequestedFor(urlMatching(url)))));
+        Helpers.verifyUserAgent();
+    }
 
     @Test
-    public void testStartArchive() throws OpenTokException {
+    public void testListArchivesWithSessionIdOffSetCount() throws OpenTokException {
         String sessionId = "SESSIONID";
+        String url = archivePath + "?offset=1&count=1&sessionId=" + sessionId;
+        stubFor(get(urlEqualTo(url))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody("{\n" +
+                                "          \"count\" : 60,\n" +
+                                "          \"items\" : [ {\n" +
+                                "            \"createdAt\" : 1395187930000,\n" +
+                                "            \"duration\" : 22,\n" +
+                                "            \"id\" : \"ef546c5a-4fd7-4e59-ab3d-f1cfb4148d1d\",\n" +
+                                "            \"name\" : \"\",\n" +
+                                "            \"partnerId\" : 123456,\n" +
+                                "            \"reason\" : \"\",\n" +
+                                "            \"sessionId\" : \"SESSIONID\",\n" +
+                                "            \"size\" : 2909274,\n" +
+                                "            \"status\" : \"available\",\n" +
+                                "            \"url\" : \"http://tokbox.com.archive2.s3.amazonaws.com/123456%2Fef546c5" +
+                                "a-4fd7-4e59-ab3d-f1cfb4148d1d%2Farchive.mp4?Expires=1395188695&AWSAccessKeyId=AKIAI6" +
+                                "LQCPIXYVWCQV6Q&Signature=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\"\n" +
+                                "          }]\n" +
+                                "        }")));
+        ArchiveList archives = sdk.listArchives(sessionId, 1, 1);
+        assertNotNull(archives);
+        assertEquals(1, archives.size());
+        assertEquals(60, archives.getTotalCount());
+        assertThat(archives.get(0), instanceOf(Archive.class));
+        assertEquals("ef546c5a-4fd7-4e59-ab3d-f1cfb4148d1d", archives.get(0).getId());
+        verify(getRequestedFor(urlEqualTo(url)));
+        assertTrue(Helpers.verifyTokenAuth(apiKey, apiSecret,
+                findAll(getRequestedFor(urlMatching(url)))));
+        Helpers.verifyUserAgent();
+    }
 
+    @Test
+    public void testListArchivesWithSessionId() throws OpenTokException {
+        String sessionId = "SESSIONID";
+        String url = archivePath + "?sessionId=" + sessionId;
+        stubFor(get(urlEqualTo(url))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody("{\n" +
+                                "          \"count\" : 60,\n" +
```

Comment:
Since this test is only returning 6 items we should change the count to 6 and compare against 6

Reply:
The count is the number of archives for that API key. 
The records returned (based on session id) can be less/same.